addons/stock_account/models/stock.py

```diff
@@ -87,27 +87,54 @@ def _get_price_unit(self):
             move = self.search([('product_id', '=', self.product_id.id),
                          ('state', '=', 'done'), 
                          ('location_id.usage', '=', 'internal'), 
-                         ('location_dest_id.usage', '!=', 'internal')], order='date desc', limit=1)
+                         ('location_dest_id.usage', '!=', 'internal'),
+                         ('company_id', '=', self.company_id.id)], order='date desc', limit=1)
             if move:
                 return move.price_unit or self.product_id.standard_price
         return self.product_id.standard_price
 
     def _update_future_cumulated_value(self, value):
         self.ensure_one()
```

Comment:
negative stock in fifo: (out something you don't have in stock) do not assign a value (0), they will be compensated by this method when a receipt compensates